Description: Fix potential buffer overflow detection
 On some architectures, calculation of edLen may lead to a
 negative value when the value returned by rand() is less
 than extraDataLen away from RAND_MAX. The negative value
 passed to memset() triggers a glibc error. Forcing the
 cast of rand() to be unsigned avoid the return of a negative
 value.

Author: Louis Bouchard <louis.bouchard@canonical.com>
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/rsyslog/+bug/1464201
Forwarded: https://github.com/rsyslog/rsyslog/issues/506
---
Index: rsyslog-8.12.0/tests/tcpflood.c
===================================================================
--- rsyslog-8.12.0.orig/tests/tcpflood.c	2015-07-08 16:31:07.000000000 +0200
+++ rsyslog-8.12.0/tests/tcpflood.c	2015-09-02 14:36:56.168433982 +0200
@@ -398,7 +398,7 @@
 			}
 		} else {
 			if(bRandomizeExtraData)
-				edLen = ((long) rand() + extraDataLen) % extraDataLen + 1;
+				edLen = ((unsigned long) rand() + extraDataLen) % extraDataLen + 1;
 			else
 				edLen = extraDataLen;
 			memset(extraData, 'X', edLen);
